openfl/component/assigner/static_grouped_assigner.py

```diff
@@ -42,6 +44,7 @@ def __init__(self, task_groups, **kwargs):
         self.task_groups = task_groups
         super().__init__(**kwargs)
 
+    @task_group_filtering
     def define_task_assignments(self):
```

Comment:
I suggest renaming the decorator to `@with_selected_task_groups`. This way the code "reads" almost like in English: "With the selected task groups, define the task assignments...". WDYT?